parser_library/src/workspaces/workspace.cpp

```diff
@@ -231,9 +230,10 @@ struct workspace_parse_lib_provider final : public parse_lib_provider
         a.register_stmt_analyzer(&hc_analyzer);
 
         co_await a.co_analyze();
-        a.collect_diags();
+        auto d = a.diags();
 
-        macro_pfc.m_last_results->macro_diagnostics = std::move(a.diags());
+        macro_pfc.m_last_results->macro_diagnostics.assign(
+            std::make_move_iterator(d.begin()), std::make_move_iterator(d.end()));
```

Comment:
Out of curiosity, is there something wrong in doing the following?

```suggestion
        macro_pfc.m_last_results->macro_diagnostics.assign(d.begin(), d.end());
```

Assuming here, that `d` is still a `std::span`.

The same goes for line 101 above and a few instances of this in `parser_library/test/async_macro_parsing.cpp`

Reply:
I will make copies instead of moves.